surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,331 @@
+One of the [compiler team's focuses for
+2022](https://blog.rust-lang.org/inside-rust/2022/02/22/compiler-team-ambitions-2022.html)
+is to improve the experience of debugging Rust code. We'd like to better
+understand your experiences using a debugger with Rust - what's been good,
+what's been bad and how your experience debugging Rust could be better..
+
+Unless you choose to enter your email, your answers will be anonymous. Any
+personal data you submit as a part of this survey will be handled in accordance
+with our policy as described in our Frequently Asked Questions:
+
+https://github.com/rust-community/team/wiki/State-of-the-Rust-Language-Community-Survey-FAQ
+
+We estimate it will take about ? minutes to complete.
+
+### Which operating systems do you use regularly for Rust debugging?
+Note: this is specifically about which systems you use for development and
+debugging not all the systems you target.
+
+Type: select all that apply (optional)
+
+- Linux
+- Windows
+- Windows Subsystem for Linux
+- Mac OS
+- Embedded devices
+- Other (open response)
+
+> *Justification:* Cohort analysis
+>
+> Debuggers are often platform-specific and various improvements that the
+> compiler team could make could only benefit a subset of all platforms, so
+> knowing what users are using is valuable.
+
+### How would you rate your Rust expertise?
+Type: select one
+
+- I can't read or write Rust
+- I can write simple programs in Rust
+- I can write useful, production-ready code but it is a struggle
+- I am productive writing Rust
+- I am an expert
+
+> *Justification:* Cohort analysis
+>
+> Users with different levels of experience are likely to face different
+> challenges.
+
+
+### What forms of debugging do you typically use?
+Type: select all that apply
+
+- `println!`/`log`/`tracing`
+- Stack traces
+- Traditional debuggers
+- Record/replay debuggers
+- High-level tracing
+
+> *Justification:*
+>
+> Understand how users are debugging Rust code.
+
+### What types of application are you debugging?
+Type: select all that apply (optional)
+
+- Software
+- Embedded hardware
+- ...
+
+> *Justification:*
+>
+> Some debugger use cases are very different from "standard" debugging.
+
+### Do you debug remotely?
+i.e. are you debugging something that's running on an embedded device/another
+platform, or something like that
+
+Type: yes/no
+
+> *Justification:*
+>
+> Remote debugging will have a different user experience that might be better
+> or worse.
+
+### When do you start debugging?
+Type: select all that apply
+
+- Fixing a crash/panic/access violation
+- Fixing a logic bug
+- Learning how software works
+- Understand runtime semantics
+- Understand a third-party library's behaviour
+- Understand own program's behaviour
+- Profiling
+
+> *Justification:*
+>
+> Different motivations for debugging will benefit from different features from
+> debuggers and might have a better or worse experience.
+
+
+### How would you rate your expertise with debuggers?
```

Comment:
Is this question about debuggers in any language/ecosystem or specifically with Rust? (If any language/ecosystem, we should probably call that out.)

Reply:
I've clarified that this is "with Rust" specifically, can add another question if we're interested more generally?